File: src/main/java/kr/co/lunatalk/domain/auth/domain/RefreshToken.java
package kr.co.lunatalk.domain.auth.domain;

import jakarta.persistence.Id;
import lombok.Builder;
import lombok.Getter;
import org.springframework.data.redis.core.RedisHash;
import org.springframework.data.redis.core.TimeToLive;

@Getter
@RedisHash(value = "refreshToken")
public class RefreshToken {

	@Id
	private Long id;

	private String refreshToken;

	@TimeToLive
	private Long ttl;

	@Builder
	public RefreshToken(Long id, String refreshToken, Long ttl) {
		this.id = id;
		this.refreshToken = refreshToken;
		this.ttl = ttl;
	}

	public static RefreshToken of(Long memberId, String refreshToken, Long ttl) {
		return RefreshToken.builder().id(memberId).refreshToken(refreshToken).ttl(ttl).build();
	}
}

File: src/main/kotlin/kr/co/lunatalk/domain/auth/domain/RefreshToken.kt
package kr.co.lunatalk.domain.auth.domain

import org.springframework.data.annotation.Id
import org.springframework.data.redis.core.RedisHash
import org.springframework.data.redis.core.TimeToLive

@RedisHash(value = "refreshToken")
class RefreshToken(
    @Id
    val id: Long,
    val refreshToken: String,
    @TimeToLive
    val ttl: Long,
) {
    companion object {
        fun of(memberId: Long, refreshToken: String, ttl: Long): RefreshToken =
            RefreshToken(id = memberId, refreshToken = refreshToken, ttl = ttl)
    }
}
